core/ir/riscv64/instr.c

```diff
@@ -120,46 +162,40 @@ bool
 instr_is_call_arch(instr_t *instr)
 {
     int opc = instr_get_opcode(instr);
-    return (opc == OP_jal || opc == OP_jalr || opc == OP_c_j || opc == OP_c_jal ||
-            opc == OP_c_jr || opc == OP_c_jalr);
+    return ((opc == OP_jal || opc == OP_jalr || opc == OP_c_jal || opc == OP_c_jalr) &&
+            opnd_get_reg(instr_get_dst(instr, 0)) != DR_REG_ZERO);
```

Comment:
The condition is `!= DR_REG_ZERO`, for `c.jal / c.jalr`, that's always true. But since we're including all the implicit operands, I think it's nicer to treat them just as the expanded one. i.e. I think it's better to write `(opc == OP_jal || opc == OP_c_jal) && dst0 != ZERO` than `(opc == OP_jal && dst0 != ZERO) || opc == OP_c_jal`.